src/Deserializers/DataValueDeserializer.php

```diff
@@ -15,45 +16,56 @@
  *
  * @license GPL-2.0+
  * @author Jeroen De Dauw < jeroendedauw@gmail.com >
+ * @author Thiemo Mättig
  */
 class DataValueDeserializer implements DispatchableDeserializer {
 
 	const TYPE_KEY = 'type';
 	const VALUE_KEY = 'value';
 
 	/**
-	 * @var string[] Associative array of data type id => DataValue class name.
+	 * @var array Associative array mapping data type IDs to either callables returning new
+	 *  DataValue objects, or full qualified DataValue class names.
```

Comment:
What do the callbacks get as a parameter? If it's just a class name, what are the expectations? I assume such classes must have a static newFromArray method?